### PR TITLE
create model, repository, service

### PR DESCRIPTION
**Pull request by OleksandrSaven**

### CODE REVIEW

**Review by ArtemReizer — APPROVED** [review]
Looks good) GJ!

**Review by Visqe — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -22,21 +22,47 @@
...
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>jakarta.persistence</groupId>
+            <artifactId>jakarta.persistence-api</artifactId>
```
Comment (vovasalyha) [inline pom.xml]: This dependency is added transitively by data-jpa starter

File: pom.xml [hunk 2]
```diff
@@ -22,21 +22,47 @@
...
+            <version>3.1.0</version>
+        </dependency>
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>1.18.24</version>
```
Comment (vovasalyha) [inline pom.xml]: Lombok version is present in parent pom

File: pom.xml [hunk 3]
```diff
@@ -22,21 +22,47 @@
...
+        </dependency>
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.33</version>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+            <version>3.1.0</version>
```
Comment (vovasalyha) [inline pom.xml]: This version also present in parent pom

File: src/main/java/com/app/onlinebookstore/OnlineBookStoreApplication.java [hunk 4]
```diff
@@ -1,13 +1,38 @@
...
+    @Autowired
+    private BookService bookService;
 
     public static void main(String[] args) {
         SpringApplication.run(OnlineBookStoreApplication.class, args);
     }
 
+    @Bean
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/OnlineBookStoreApplication.java]: Better use lambda.

File: src/main/java/com/app/onlinebookstore/model/Book.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import java.math.BigDecimal;
+import lombok.Data;
+
+@Entity
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/model/Book.java:12]: Add `Table` annotation too.

File: src/main/java/com/app/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> getAllBooksQuery = session.createQuery("from Book ", Book.class);
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/repository/impl/BookRepositoryImpl.java]:
```suggestion
            Query<Book> getAllBooksQuery = session.createQuery("FROM Book ", Book.class);

```

File: src/main/java/com/app/onlinebookstore/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
```
Comment (vovasalyha) [inline src/main/java/com/app/onlinebookstore/service/impl/BookServiceImpl.java:15]: You can use lombok annotation to generate constructor

File: src/main/resources/application.properties [hunk 8]
```diff
@@ -1 +1,7 @@
-
+spring.datasource.url=jdbc:mysql://localhost:3306/bookapp?serverTimezone=UTC
+spring.datasource.username=root
+spring.datasource.password=gjujlf123$_0
```
Comment (vovasalyha) [inline src/main/resources/application.properties:3]: I hope you are not using this password for your real account :)

**Review by vovasalyha — CHANGES_REQUESTED** [review]
GJ. Let's improve few things.

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**